memory/memcached.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+
+import time
+import getpass
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import memory
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Memcached(Test):
+
+    """
+    Memcached - High performance memory object caching system.
+    This test Runs the memcached server in the background, based
+    upon the memory param given. And it tries to run the memcslap
+    stress tool which generates a load against memcached server.
+    For more options on memcached:
+    Refer - https://linux.die.net/man/1/memcached
+    """
+
+    def setUp(self):
+
+        """
+        Sets up the args required to run the test.
+        """
+
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+
+        if detected_distro.name not in ['Ubuntu', 'redhat']:
+            self.skip('Test Not applicable')
+
+        if detected_distro.name == "Ubuntu":
+            deps = ['memcached', 'libmemcached-tools']
+            stress_tool = 'memcslap'
+
+        if detected_distro.name == "redhat":
```

Comment:
As of now verified in redhat ...may be later support for the distros can be verfified and added.